share/admin.py

```diff
@@ -181,13 +183,37 @@ class AccessTokenAdmin(admin.ModelAdmin):
 
 
 class ProviderRegistrationAdmin(admin.ModelAdmin):
-    list_display = ('source_name', 'status_', 'submitted_at', 'submitted_by', 'direct_source')
-    list_filter = ('direct_source', 'status',)
+    list_display = ('source_name', 'status_', 'submitted_at', 'submitted_by', 'source_oai', 'direct_source')
+    list_filter = ('direct_source', 'status', 'source_oai')
     readonly_fields = ('submitted_at', 'submitted_by',)
+    actions = ('create_push_account',)
 
     def status_(self, obj):
         return ProviderRegistration.STATUS[obj.status].title()
 
+    def create_push_account(self, request, queryset):
+        for submission in queryset:
+            if not submission.direct_source:
+                messages.error(request, '{} is not a direct source.'.format(submission.source_name))
+                continue
+
+            source_url = re.compile('(\.|://)').split(str(submission.source_base_url))
```

Comment:
Could you convert the bulk of this into a method on the registration object itself and then add a test for it?